### PR TITLE
JBTM-2308 Fixes for extra JTS bean types

### PR DESCRIPTION
**Pull request by mmusgrov**
!XTS !QA_JTA !QA_JTS_JACORB !QA_JTS_JDKORB !PERF !BLACKTIE

### CODE REVIEW

File: ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+import com.arjuna.ats.arjuna.common.Uid;
+import com.arjuna.ats.arjuna.coordinator.AbstractRecord;
+import com.arjuna.ats.arjuna.coordinator.BasicAction;
+import com.arjuna.ats.arjuna.coordinator.RecordList;
+import com.arjuna.ats.arjuna.tools.osb.mbean.*;
+import com.arjuna.ats.internal.jts.recovery.transactions.RecoveredTransaction;
+
+public class RecoveredTransactionWrapper extends RecoveredTransaction implements ActionBeanWrapperInterface {
+    UidWrapper wrapper;
```
Comment (gytis) [inline ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java]: Shouldn't these be private? (Same for the other wrapper class)

File: ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    ActionBean action;
+    boolean activated;
```
Comment (gytis) [inline ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java]: Shouldn't these be private? (Same for the other wrapper class)

File: ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import com.arjuna.ats.arjuna.common.Uid;
+import com.arjuna.ats.arjuna.coordinator.AbstractRecord;
+import com.arjuna.ats.arjuna.coordinator.BasicAction;
+import com.arjuna.ats.arjuna.coordinator.RecordList;
+import com.arjuna.ats.arjuna.tools.osb.mbean.*;
+import com.arjuna.ats.internal.jts.recovery.transactions.RecoveredTransaction;
+
+public class RecoveredTransactionWrapper extends RecoveredTransaction implements ActionBeanWrapperInterface {
+    UidWrapper wrapper;
+    ActionBean action;
```
Comment (gytis) [inline ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java]: wrapper and action do not seem to be used. (Same for the other wrapper class)

**Comment by gytis** [comment]
Seems that tab is used for indentation in many places rather than spaces.

File: ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java (hunk 2 above)
Comment (mmusgrov) [inline ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java]: These classes are instantiated by reflection:

{code}
    Class<ActionBeanWrapperInterface> cl = (Class<ActionBeanWrapperInterface>) Class.forName(w.getClassName());
    Constructor<ActionBeanWrapperInterface> constructor = cl.getConstructor(ActionBean.class, UidWrapper.class);
    ra = constructor.newInstance(this, w);
{code}

so if I were to mark the constructor as private it produce IllegalAccessException exceptions

File: ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java (hunk 3 above)
Comment (mmusgrov) [inline ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java]: see my previous comment, they're instantiated by reflection.

File: ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java (hunk 2 above)
Comment (gytis) [inline ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java]: I mean class attributes, not constructor.

**Comment by mmusgrov** [comment]
We agreed that style should follow whatever format was already in effect. As I recall the file you refer to already used a mix of spaces and tabs.

**Comment by mmusgrov** [comment]
My IDE (intellij) has "Use tab character" unchecked but I will investigate.

File: ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java (hunk 2 above)
Comment (mmusgrov) [inline ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/tools/osb/mbean/jts/RecoveredTransactionWrapper.java]: OK I'll go back and update their access control (and fix the tab at the same time)

**Comment by gytis** [comment]
Fair enough. If it was indented with tabs before, then I think it's correct to use them in the update. However, it doesn't seem consistent in the newly added lines, some are with spaces others are with tabs.

**Comment by mmusgrov** [comment]
The last commit marks the local fields of the two new wrappers private

**Comment by mmusgrov** [comment]
MAIN profile tests failure on mac is unrelated

**Pull request closed without merging**